eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/jpa/GapAwareTrackingTokenOperations.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore.jpa;
+
+import org.axonframework.common.DateTimeUtils;
+import org.axonframework.eventhandling.GapAwareTrackingToken;
+import org.axonframework.eventhandling.GenericEventMessage;
+import org.axonframework.eventhandling.TrackingToken;
+import org.slf4j.Logger;
+
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+/**
+ * Contains operations that are used to interact with {@link GapAwareTrackingToken} used in Aggregate based JPA event
+ * store implementation.
+ *
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ */
+record GapAwareTrackingTokenOperations(
```

Comment:
Can you help me explain why you made several of these "Operations" records?

Reply:
All of them as you said - it was easier for me to do the new implementation step-by-step extracting common parts.
We can delete them after deletion of the old one, but I'm not sure about that. Currently, all of these operations record are more consistent, easier to understand than everything in one class. I needed to skip to many places before in order to grasp what we're doing with the gap aware token.

Reply:
But, wouldn't that make it a feature of the `GapAwareTrackingToken` to be cleaned, instead of another file for this single operation?